packages/portal-proto/src/features/cDave/CDaveCard/ContinuousData.tsx

```diff
@@ -161,13 +164,14 @@ const ContinuousData: React.FC<ContinuousDataProps> = ({
           .map(({ key }) => key)
           .slice(0, 2),
       );
+    }
 
     if (customBinnedData === null) {
       setYTotal(displayedData.reduce((a, b) => a + b.count, 0));
     }
 
     setSelectedFacets([]);
-  }, [displayedData, customBinnedData]);
+  }, [dataDimension, displayedData, customBinnedData]);
```

Comment:
We don't want to update the survival plot when the dataDimension changes. So I am using ref to check that. 

If we just add `dataDimension` to the dep array, it will trigger the survival plot changes which we don't want.

Does this make sense? Also, do you think there's a better way to achieve this?

Reply:
Ah, I see. I think this solution is a little confusing and prone to resulting in bugs where the things don't update when we expect them to. `dataDimension` represents whether the card is displaying years vs days, so I'm not sure how I get how this is fixing the issue described in the ticket. I think we probably need a useEffect looking for changes on customBinned data that resets selectedSurvivalPlots to an empty array.

Reply:
Ah, I see. I think this solution is a little confusing and prone to resulting in bugs where the things don't update when we expect them to. `dataDimension` represents whether the card is displaying years vs days, so I'm not sure how I get how this is fixing the issue described in the ticket. I think we probably need a useEffect looking for changes on customBinned data that resets selectedSurvivalPlots to an empty array.

Reply:
I see how this might be confusing. I previously `[worked](https://github.com/NCI-GDC/gdc-frontend-framework/pull/1119)` on a fix for this [ticket](https://gdc-ctds.atlassian.net/browse/PEAR-1852) to prevent the survival plot from updating when dataDimension changes. However, my earlier changes introduced a new issue reported in PEAR-1974. To address this, I'm proposing a new solution for PEAR-1852 that avoids unintended side effects.